routers/api/v1/repo/mirror.go

```diff
@@ -42,6 +43,11 @@ func MirrorSync(ctx *context.APIContext) {
 		ctx.Error(http.StatusForbidden, "MirrorSync", "Must have write access")
 	}
 
+	if !setting.Mirror.Enabled {
+		ctx.Status(http.StatusBadRequest)
```

Comment:
No additional message? Just a `400` return code?
I think a message would be appropriate here so that clients know it is not their fault if a request failed, which is normally implicated by a `4XX` request…